### DISCUSSION TITLE
How to get address of Object<FungibleAsset>

### DISCUSSION THREAD

**Discussion by Levi0804** [aptos-labs/aptos-developer-discussions#560] [General]
Hello, I wanted to know if there's a way to get address of `Object<FungibleAsset>`. One possible approach could have been using `object_address<T>()` which requires `T: key` but here `FungibleAsset` doesn't implement key, I was wondering if there are any other ways to get the object's address.

**Comment by kshitijc1506** [comment]
`FungibleAsset` is an ephemeral struct, you access it after a withdrawal and it must be consumed in a deposit. This will be checked at compile time. 

If you're looking for the asset's address, you're looking for the Metadata object address. You can use `object_address` to retrieve it. 

```rust
object::object_address(&Object<Metadata>)
```

**Reply by kshitijc1506** [reply]
If you created the asset in a smart contract you deployed then you can also retrieve it like [this](https://github.com/aptos-labs/aptos-core/blob/94c4f1ec39ef1f9d688ff892651e2cba36f021e2/aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move#L88)

**Reply by Levi0804** [reply]
thanks a lot!
